public/components/trace_analytics/components/common/legacy_route_helpers.ts

```diff
@@ -12,3 +17,72 @@ export const convertLegacyTraceAnalyticsUrl = (location: Location) => {
   }`;
   return pathname + hash;
 };
+
+export const convertTraceAnalyticsNewNavUrl = (location: Location) => {
+  const pathname = location.pathname;
+  const hash = location.hash;
+
+  const isNewNavEnabled = coreRefs?.chrome?.navGroup?.getNavGroupEnabled();
+
+  // Handle service URLs with IDs
+  if (hash.includes('#/services/')) {
+    const serviceId = location.hash.split('/services/')[1]?.split('?')[0] || '';
+    if (serviceId) {
+      if (isNewNavEnabled) {
+        window.location.assign(
+          `/app/${observabilityServicesNewNavID}#/services?datasourceId=&serviceId=${encodeURIComponent(
+            serviceId
+          )}`
+        );
+      } else {
+        window.location.assign(
+          `/app/${observabilityTracesID}#/services?datasourceId=&serviceId=${encodeURIComponent(
+            serviceId
+          )}`
+        );
+      }
+      return;
+    }
+  }
```

Comment:
URL query parameters are not ordered by default. Browsers, servers, and other systems generally treat query parameters as unordered key-value pairs.

```
https://example.com/page?param1=value1&param2=value2
https://example.com/page?param2=value2&param1=value1
```
With that said we don't need to see if MDS is enabled. The url parser on load should default to `dataSource=""` if datasource is not defined.

Reply:
This check is for if a id is provided. The new navigation check is to handle the different urls used for both
observabilityServicesNewNavID vrs observabilityTracesID
This isn't affected by if MDS is enabled, as the legacy re-direction should only be for local host id's.